commit 074585b3e3b0d898cc6a2d61a206e7b9bf16796d
Author: Xavier Claessens <xavier.claessens@collabora.co.uk>
Date:   Tue May 21 15:12:03 2013 +0200

    GtkMenuItem: Take padding into account to position submenu arrow
    
    Since commit a69619edbc2b36aadbff200918ea9f11885d6109, it not
    possible anymore for the style to add space between the border
    and the arrow.
    
    https://bugzilla.gnome.org/show_bug.cgi?id=700779

diff --git a/gtk/gtkmenuitem.c b/gtk/gtkmenuitem.c
index 73c2545..e54165d 100644
--- a/gtk/gtkmenuitem.c
+++ b/gtk/gtkmenuitem.c
@@ -1665,12 +1665,12 @@ gtk_menu_item_draw (GtkWidget *widget,
 
       if (direction == GTK_TEXT_DIR_LTR)
         {
-          arrow_x = x + w - arrow_size;
+          arrow_x = x + w - arrow_size - padding.right;
           angle = G_PI / 2;
         }
       else
         {
-          arrow_x = x;
+          arrow_x = x + padding.left;
           angle = (3 * G_PI) / 2;
         }
 
